### ISSUE TITLE
Could not authenticate you. Response code: 401

### ISSUE THREAD

**Issue by FranG** [nitsan-technologies/ns_twitter#29]
The extension generates an error:
Could not authenticate you. Response code: 401

**Comment by mightynico** [comment]
same here. PHP 8.1/Typo3 11.5.27

**Comment by nitsan-technologies** [comment]
Hey Guys!

Sorry, but there seem to be major breaking changes and deprecation from Twitter API. Unfortunately, there is no official documentation, e.g., We might be required to migrate from API v1 to v2 :(

**References**

- https://www.engadget.com/twitter-shut-off-its-free-api-and-its-breaking-a-lot-of-apps-222011637.html
- https://twittercommunity.com/t/401-unauthorized-32-could-not-authenticate-you/166636/9
- https://twittercommunity.com/t/error-authorizing-to-twitter-api-1-0-getting-a-401-error-with-the-json-response-error-being-32-could-not-authenticate-you/171229

Unfortunately, Due to our internal overhead workload, we don't have time to research and develop :(

Cheers, Team NITSAN

**Comment by mschmitz89** [comment]
Hey, are there any news for the update to Twitter API v2?

**Comment by nitsan-technologies** [comment]
Hey there,

Thank you for your interest in our  extension.

Our exploration of the Twitter API v2 led us to utilize endpoints exclusively accessible through the acquisition of a Twitter API Key. However, due to alterations in Twitter's policy, these API endpoints are now solely accessible to users with a purchased API key. Consequently, the existing functionality in this extension is no longer operational.

In conclusion, a comprehensive redevelopment of all API features in this extension is necessary. If urgency is a factor, you may consider sponsoring the work. For further details or expressions of interest, kindly reach out to us at https://t3planet.com/contact.

Cheers, Team NITSAN
